tests/test_lock.c

```diff
@@ -150,7 +172,35 @@ test_one_session(void **state)
     /* unlock last locked module */
     ret = sr_unlock(sess, "when1");
     assert_int_equal(ret, SR_ERR_OK);
+    
+    /* start a candidate datastore session */
+    sess = NULL;
+    ret = sr_session_start(st->conn, SR_DS_CANDIDATE, &sess);
+    assert_int_equal(ret, SR_ERR_OK);
+    
+    /* lock all modules */
+    ret = sr_lock(sess, NULL);
+    assert_int_equal(ret, SR_ERR_OK);
+    
+    /* create a candidate datastore file in /dev/shm */
+    ret = asprintf(&path, "%s/sr_%s.candidate", SR_SHM_DIR, "test");
+    assert_int_not_equal(ret, -1);   
+    ret = create_file(path, 0666);
+    assert_int_equal(ret, SR_ERR_OK);
```

Comment:
Please, do not use similar hacks, there is no need for them nor for an additional `utils.c` file for that matter. You can simply set some changes in the `candidate` datastore and it will then be created.